### DISCUSSION TITLE
Construct from dict/mapping.

### DISCUSSION THREAD

**Discussion by psippola** [pydantic/pydantic#4597] [Feature Request]
Could it be a convenient ability to construct model instances directly from a dict (or any other mapping type), without having to unpack the dict?

In Python>=3.8, with the aid of positional-only parameters, this could be achieved by changing the signature of `BaseModel.__init__` from
```python
def __init__(__pydantic_self__, **data: Any):
```
to
```python
def __init__(self, data: Mapping = _missing, /, **kwargs: Any):
```
and then combining `data` with `kwargs` before validation, if given. With an additional check, `data` could also be an iterable with field values listed in order.

- Benefits: 

  Initialising from a dict could be as simple as
  
  ```python
  from pydantic import BaseModel
  
  class User(BaseModel):
      id: int
      name: str
  
  data = {"id": 1, "name": "John Doe"}
  
  user = User(data)
  
  ```
  
  Also, updating the data partially could happen very nicely:
  
  ```python
  user = User(data, name="Jane")
  ```

  This is actually much like plain python dicts behave with different arguments.

- Disadvantages:
  I could not imagine any, since plain pydantic models do not accept positional arguments, so any existing code would not break.

**Comment by RouvenG** [comment]
You can achieve this by using

```user = User(**data)```

**Comment by Viicos** [comment]
You can now use [`BaseModel.model_validate`](https://docs.pydantic.dev/latest/usage/models/#helper-functions):

```python
from pydantic import BaseModel

class User(BaseModel):
    id: int
    name: str

data = {"id": 1, "name": "John Doe"}

user = User.model_validate(data)
```
